src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -71,20 +113,87 @@ public void stop() {
     spinWheels(0);
   }
 
+  public void hold() {
+    spinWheels(isConeMode ? kConeIntakeSpeed : kCubeIntakeSpeed);
+  }
+  
+  public Command pullTimed(double seconds, boolean isConeMode){
+    return new StartEndCommand(this::pull, this::stop, this).withTimeout(seconds).beforeStarting(isConeMode ? this::setConeMode : this::setCubeMode);
+  }
+
+  public Command pushTimed(double seconds, boolean isConeMode){
+    return new StartEndCommand(this::push, this::stop, this).withTimeout(seconds).beforeStarting(isConeMode ? this::setConeMode : this::setCubeMode);
+  }
+
   @Override
   public void close() throws Exception {
     // This method will close all device handles used by this object and release any other dynamic memory.
     // Mostly for JUnit tests
     motor.close();
     motor2.close();
-    
+  }
+
+  public void changeFlightState() {
+    // Check distances
+    int coneDistance = timeOfFlightSensor.getDistance0();
+    int cubeDistance = timeOfFlightSensor.getDistance1();
+
+    // Log distance values
+    coneDistLog.log((double)coneDistance);
+    cubeDistLog.log((double)cubeDistance);
+
+    // Check if sensors are online
+    boolean coneSensorUp = coneDistance != -1;
+    boolean cubeSensorUp = cubeDistance != -1;
+
+    // Log whether sensors are online
+    sensor0Up.setBoolean(coneSensorUp);
+    sensor1Up.setBoolean(cubeSensorUp);
+
+    // Log if sensors are activated
+    hasConeLog.log(coneDistance <= mmConeActivationThreshold && coneSensorUp);
+    hasCubeLog.log(cubeDistance <= mmCubeActivationThreshold && cubeSensorUp);
+
+    // Change state (only if sensors are online)
+    switch(tofState) {
+      case IDLE:
+        if (coneSensorUp && coneDistance <= mmConeActivationThreshold) {
+          tofState = FlightStates.CONE;
+          break;
+        } else if (cubeSensorUp && cubeDistance <= mmCubeActivationThreshold) {
+          tofState = FlightStates.CUBE;
+          break;
+        }
+      case CONE:
+        if (coneSensorUp && coneDistance > mmConeActivationThreshold) {
+          tofState = FlightStates.IDLE;
+          break;
+        }
+      case CUBE:
+        if (cubeSensorUp && cubeDistance > mmCubeActivationThreshold) {
+          tofState = FlightStates.IDLE;
+          break;
+        }
+      case CONE_SCORE:
+        if (coneSensorUp && coneDistance > mmConeActivationThreshold) {
+          tofState = FlightStates.IDLE;
+          break;
+        }
+    }
+  }
```

Comment:
Put the breaks add the end of each case (outside of any if-blocks). Otherwise it will fall through to the next case if the conditions aren't met.